### DISCUSSION TITLE
artillery called from a nodejs app

### DISCUSSION THREAD

**Discussion by luisfilsilva** [artilleryio/artillery#1240] [Artillery]
Hi experts,

Not sure if this is a strange question for you, but.. is it possible to launch artillery inside a nodejs application, instead of using a command line? I wanted to expose artillery in my app that listens for websocket commands (start load test, share stats, etc)..

Right now, I'm able to make it work by using the module spawn, but was wondering if there is another approach.

const artillery = spawn("./node_modules/.bin/artillery", [
      "run",
      "--target",
      message.target,
      "--overrides",
      "{\"config\": { \"phases\": [{ \"duration\": "+message.duration+", \"arrivalRate\": "+message.rate+", \"name\": \"Load Test\" }] } }",
      "-k",
      "config.yml",
    ]);

    artillery.stdout.on("data", (data) => {
      console.log(`stdout: ${data}`);
      socket.emit("message", data + "");
    });

Thanks!!
Luís

**Comment by hassy** [comment]
hi @luisfilsilva 👋  yes, the way you're doing it is the general recommended route, i.e. wrap the `artillery` binary as you would any other binary. (we don't recommend trying to use Artillery as a Node.js library via `require()`)

**Reply by luisfilsilva** [reply]
Thanks Hassy!! Quick as usual!! Will continue using this approach then! Regards, Luís
